frankenstein/cmd/frankenstein/main.go

```diff
@@ -86,7 +86,9 @@ func main() {
 	case distributor:
 		setupDistributor(consul, consulPrefix, chunkStore, remoteTimeout)
 	case ingestor:
-		setupIngestor(consul, consulPrefix, chunkStore, listenPort)
+		ingestor := setupIngestor(consul, consulPrefix, chunkStore, listenPort)
+		defer deleteIngestorConfigFromConsul(consul, consulPrefix)
+		defer ingestor.Stop()
```

Comment:
Doesn't this need a signal handler? Like in https://github.com/prometheus/prometheus/blob/master/cmd/prometheus/main.go#L193-L202

When stopping a Go program with SIGTERM (not sure what k8s sends) without a signal handler, the deferred functions are not executed.